### PR TITLE
Clearer tick generation on Axis.Time

### LINKED ISSUE #1606: "endless october" time axis bug
second-to-last ticks don't appear, which makes the last label seem to cover much more space than it should.
Endless Friday the 10th: 
screen shot 2015-02-09 at 4 59 50 pm

Endless left-side
screen shot 2015-02-09 at 5 04 11 pm

http://jsfiddle.net/rndeLxo6/

### PR DESCRIPTION
**Pull request by jtlan**
(from an earlier state of **develop**)

We were splicing/etc, when the correct tick marks were already generated.

Close #1606.

### CODE REVIEW

**Comment by jtlan** [comment]
Retarget of #1607 FROM THE PAST

**Comment by crmorford** [comment]
looks just as good!

File: src/components/axes/timeAxis.ts [hunk 1]
```diff
@@ -427,9 +397,10 @@ export module Axis {
         this._cleanTier(i);
       }
 
-      var tierTicks = tierConfigs.map((config: TimeAxisTierConfiguration, i: number) =>
-        this._renderTierLabels(this._tierLabelContainers[i], config, i)
-      );
+      var tierTicks = tierConfigs.map((config: TimeAxisTierConfiguration, i: number) => {
+        this._renderTierLabels(this._tierLabelContainers[i], config, i);
+        return this._getTickValuesForConfiguration(config);
+      });
```
Comment (bluong) [inline src/components/axes/timeAxis.ts]: How much cleaner would it be to split this into a foreach and a map?  Not that big of a deal

**Comment by bluong** [comment]
Code removal!

File: src/components/axes/timeAxis.ts [hunk 2]
```diff
@@ -306,10 +306,16 @@ export module Axis {
...
+      tickPos.unshift(domain[0]);
+      tickPos.push(domain[1]);
+      return tickPos;
+    }
+
+    private _renderTierLabels(container: D3.Selection, config: TimeAxisTierConfiguration, index: number) {
+      var tickPos = this._getTickValuesForConfiguration(config);
       var labelPos: Date[] = [];
       if (this._tierLabelPositions[index] === "between" && config.step === 1) {
         tickPos.map((datum: any, index: any) => {
```
Comment (bluong) [inline src/components/axes/timeAxis.ts]: Not in the scope of this PR but why is there a map usage here...

**Pull request merged**